src/Aspire.ProjectTemplates/templates/aspire-empty/AspireApplication.1.ServiceDefaults/Extensions.cs

```diff
@@ -104,14 +104,19 @@ public static WebApplication MapDefaultEndpoints(this WebApplication app)
         // Uncomment the following line to enable the Prometheus endpoint (requires the OpenTelemetry.Exporter.Prometheus.AspNetCore package)
         // app.MapPrometheusScrapingEndpoint();
 
-        // All health checks must pass for app to be considered ready to accept traffic after starting
-        app.MapHealthChecks("/health");
-
-        // Only health checks tagged with the "live" tag must pass for app to be considered alive
-        app.MapHealthChecks("/alive", new HealthCheckOptions
+        // Adding health checks endpoints to applications in non-development environments has security implications.
+        // See https://aka.ms/dotnet/aspire/healthchecks for details before enabling these endpoints in non-development environments.
+        if (app.Environment.IsDevelopment())
```

Comment:
While this works, it feels a bit like instead of making this method a no-op we'd ideally condition the callsites in the Program.cs files for each microservice instead. I can understand why it's more convinient to do it here, but just adding the obvious comment in case we want to consider it.

Reply:
Also, should the condition wrap the commented out code for Prometheus too?

Reply:
> we'd ideally condition the callsites in the Program.cs files for each microservice instead

The problem with that is that the call to `MapDefaultEndpoints` is added by the VS tooling for enlisting in Aspire orchestration so I don't think it really scales to do it there.

> Also, should the condition wrap the commented out code for Prometheus too?

What's the thinking there? That the comment applies to both? I don't think putting the Prometheus endpoint in dev only, commented out, is that useful, but I could see that the comment about security implications of these kinds of endpoints could be useful to expand to cover Prometheus too.

Reply:
> The problem with that is that the call to MapDefaultEndpoints is added by the VS tooling for enlisting in Aspire orchestration so I don't think it really scales to do it there.

Yeah, that's what I thought. We could of course ask the VS Tooling folks to also emit the condition but that would be likely hard to do at this stage. I'm fine with this approach, I just think that in an ideal world I feel like the other one would be a bit more accurate.

> What's the thinking there? That the comment applies to both? I don't think putting the Prometheus endpoint in dev only, commented out, is that useful, but I could see that the comment about security implications of these kinds of endpoints could be useful to expand to cover Prometheus too.

Yeah, mainly thinking about security implications. Your comment above the if is specific about health checks, but technically speaking it applies to any endpoints you expose with your app, and given these are all enabled for all of your microservices by default I thought it might make sense to have the user explicitly take action to expose these in prod too.